### ISSUE TITLE
Reducer should do better at removing return statements

### ISSUE THREAD

**Issue by afd** [google/graphicsfuzz#387]
We get things like:

```
 return vec3(1.0);
 return vec3(1.0);
 return vec3(1.0);
 return vec3(1.0);
```

left over, and it's because the reducer currently is scared of making a shader invalid by removing a return.  This could be improved.

**Comment by afd** [comment]
More generally, when not preserving semantics we can be agressive about removing returns when the function is void or when there is a "post-dominating" return - i.e. a return that is guaranteed to be reached if the current return is removed.

**Comment by afd** [comment]
Fixed by #404
